### PR TITLE
CDM-87 Implemented partition-range exception file generation for Migrate

### PR DESCRIPTION
**Pull request by pravinbhat**
**What this PR does**:
Capture failed partitions in a file for easier reruns
Optimized mvn to reduce jar size
Fixed bugs in docs

**Which issue(s) this PR fixes**:
Fixes # CDM-87

**Checklist:**
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by msmygit — APPROVED** [review]

File: src/main/scala/com/datastax/cdm/job/BaseJob.scala [hunk 1]
```diff
@@ -70,15 +69,11 @@ abstract class BaseJob[T: ClassTag] extends App {
     maxPartition = getMaxPartition(propertyHelper.getString(KnownProperties.PARTITION_MAX), hasRandomPartitioner)
     coveragePercent = propertyHelper.getInteger(KnownProperties.TOKEN_COVERAGE_PERCENT)
     numSplits = propertyHelper.getInteger(KnownProperties.PERF_NUM_PARTS)
-    partitionFile = propertyHelper.getString(KnownProperties.PARTITIONS_TOKEN_RANGE_FILE)
```
Comment (faizalrub-datastax) [inline src/main/scala/com/datastax/cdm/job/BaseJob.scala:73]: Is this being handled in another class?  This exception file generated from the main job run (for each table) will be the input for rerunning jobs with with the class DiffPartitionsFromFile / MigratePartitionsFrom File.

**Review by faizalrub-datastax — APPROVED** [review]

File: src/main/scala/com/datastax/cdm/job/BaseJob.scala (hunk 1 above)
Comment (pravinbhat) [reply src/main/scala/com/datastax/cdm/job/BaseJob.scala:73]: @faizalrub-datastax This property was not used in the code anywhere. It instead uses `keyspace.tablename` to generate the file name. I have updated the logic a bit, it now generates the file as `/exception-dir/<keyspace>.<table>_partitions.csv` & this file can be used to rerun `DiffPartitionsFromFile` & `MigratePartitionsFrom` jobs. The same has been documented in the readme [here](https://github.com/datastax/cassandra-data-migrator/tree/feature/CDM-87#migrating-specific-partition-ranges). Applicable configs have also been cleaned up.

Comment (faizalrub-datastax) [reply src/main/scala/com/datastax/cdm/job/BaseJob.scala:73]: @pravinbhat for this particular customer, job runs are automated and the exception file is passed as conf parameter for the spark-submit as follows ,  --conf spark.input.partitionFile=<dir>/<filename>

Comment (pravinbhat) [reply src/main/scala/com/datastax/cdm/job/BaseJob.scala:73]: @faizalrub-datastax However that is not used in the code anywhere including `DiffPartitionsFromFile` & `MigratePartitionsFrom`. Can you check if these changes from your customer specific branches were missed & not merged in main.

**Pull request merged**